tests/integration/test.changes.js

```diff
@@ -465,7 +465,16 @@ adapters.forEach(function (adapter) {
       });
     });
 
-    it('3356 throw inside a filter', function (done) {
+    it.only('3356 throw inside a filter', function (done) {
+      var testFor5238 = function (err) {
+        done('There was an unhandledRejection ' + err);
+      };
+      testUtils.addGlobalEventListener(
+        testUtils.getUnHandledRejectionEventName(), testFor5238);
+      after(function () {
```

Comment:
I didnt know it could work like this, nice, however for consistency any time we are removing a listener to cleanup a test we do so just before calling `done` (or returning the last promise) so if we could do that, ie

        }).catch(function () {
          testUtils.removeGlobalEventListener(testUtils.getUnHandledRejectionEventName(), testFor5238);
          done();

Its a little clearer I think. Also a tiny nit but if we could shorten the addListener line, maybe 

    testUtils.addUnhandledRejectionListener(testFor5238)

And do the name switching inside testUtils, how does that sound?